arch/risc-v/src/common/crt0.c

```diff
@@ -169,13 +169,15 @@ void _start(int argc, char *argv[])
 
   ARCH_DATA_RESERVE->ar_sigtramp = (addrenv_sigtramp_t)sig_trampoline;
 
+#ifdef CONFIG_HAVE_CXX
   /* Call C++ constructors */
 
   exec_ctors();
 
   /* Setup so that C++ destructors called on task exit */
 
   atexit(exec_dtors);
```

Comment:
The problem is call to this. The fix IMO is not to mask this functionality but to fix atexit(), which is done here:
https://github.com/apache/incubator-nuttx/pull/6197

The reason for the crash, is that atexit runs the dtors in supervisor mode, and the MMU does not allow this.

Reply:
But, it's good to skip the call and register if user don't enable c++ since we can save the code size.

Reply:
Fine by me